fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java

```diff
@@ -331,22 +317,41 @@ A not in (1, 2, 3, 100):
     }
 
     @Override
-    public Statistics visitNot(Not not, EstimationContext none) {
-        Preconditions.checkState(!(not.child() instanceof Not),
-                "Consecutive Not statement should be merged previously");
-        EstimationContext context = new EstimationContext(true, none.statistics);
-        return not.child().accept(this, context);
+    public Statistics visitNot(Not not, EstimationContext context) {
+        Statistics childStats = new FilterEstimation().estimate(not.child(), context.statistics);
+        double rowCount = Math.max(context.statistics.getRowCount() - childStats.getRowCount(), 0);
+        if (rowCount == 0.0) {
+            return Statistics.zero(context.statistics);
```

Comment:
it is better to return at least one tuple, not Zero.

Reply:
Why